### PR TITLE
Mes 397 journal intergration

### PR DESCRIPTION
**Pull request by ammarhaiderbjss**
- Uses [dotenv-safe](https://www.npmjs.com/package/dotenv-safe) to manage environment variables in a .env file, provides a way of ensuring that the required environment variables have been set up by throwing exceptions if the `.env` file at build time doesn't match the keys in `.env.example`
- Creates a custom webpack config file that passes env vars along to the app
- Updates journal provider to handle real data
- Updates journal unit tests

### CODE REVIEW

File: src/mocks/data/journalResp.json [hunk 1]
```diff
@@ -1,180 +1 @@
...
...\":\"Ccc\",\"lastName\":\"White\"},\"driverNumber\":\"AAAAA123456A78AA\",\"gender\":\"Female\",\"candidateAddress\":{\"line1\":\"1 Some Street\",\"line2\":\"Some place\",\"line3\":\"Some area\",\"line4\":\"Some town\",\"line5\":\"Some county\",\"postcode\":\"AA1 2BB\"},\"primaryTelephone\":\"01234 567890\",\"secondaryTelephone\":\"04321 098765\",\"mobileTelephone\":\"07654 123456\",\"email\":\"aaaa@bbbb.com\",\"prn\":11223344,\"previousADITests\":2},\"application\":{\"_applicationId\":\"384 392 293 22\",\"_bookingSequence\":\"03\",\"_checkDigits\":\"01\",\"welshTest\":false,\"extendedTest\":true,\"meetingPlace\":\"Some meeting place\",\"progressiveAccess\":false,\"specialNeeds\":\"Heavily pregnant\",\"checkMarker\":false,\"vehicleSeats\":5,\"vehicleHeight\":1.75,\"vehicleWidth\":2.5,\"vehicleLength\":5.5,\"testCategory\":\"B+E\"},\"previousCancellations\":[\"Act of nature\",\"DVSA\",\"DSA\"],\"business\":{\"_id\":1000,\"name\":\"Dave's Dodgy Driving Instructors\",\"businessAddress\":{\"line1\":\"Flat 1A\",\"line2\":\"1 Example Street\",\"line3\":\"Example Area\",\"line4\":\"Example Town\",\"line5\":\"Example County\",\"postcode\":\"AB12 3CD\"},\"telephone\":\"07890 654321\"}}}]}}"}
```
Comment (zielinskikamil) [inline src/mocks/data/journalResp.json]: do we really want to keep not formatted JSON in repo?

Comment (danielchildsbjss) [inline src/mocks/data/journalResp.json]: Do we want to do this? Looks hard to edit now!

File: src/providers/app-config/app-config.ts [hunk 2]
```diff
@@ -17,9 +21,21 @@ export class AppConfigProvider {
     backgroundColor: '#ffffff'
   };
 
-  constructor() {}
+  constructor() {
+    this.journalApiUrl = this.readString(EnvVarKeys.JOURNAL_API_URL, 'assets/data/journal-data.json');
+  }
+
+  getJournalApiUrl(): string {
+    return this.journalApiUrl;
```
Comment (danielchildsbjss) [inline src/providers/app-config/app-config.ts]: Might want to calculate this each time so we can change at runtime?

File: src/providers/journal/journal.spec.ts [hunk 3]
```diff
@@ -15,22 +16,22 @@ describe('PROVIDER: journal - ', () => {
   };
 
   beforeEach(() => {
-    provider = new JournalProvider(httpMock as any, dateTimeUtilMock);
+    provider = new JournalProvider(httpMock as any, dateTimeUtilMock, new AppConfigProvider());
```
Comment (danielchildsbjss) [inline src/providers/journal/journal.spec.ts]: We should mock out AppConfigProvider for unit tests

File: src/mocks/data/journalResp.json (hunk 1 above)
Comment (ammarhaiderbjss) [reply src/mocks/data/journalResp.json]: No we don't, but this is the exact format of the response you get from the microservice. Now that the app is integrated it needs to follow the same format, I'll create a task to clean this up in both repo's at a later time..

Comment (ammarhaiderbjss) [reply src/mocks/data/journalResp.json]: No we don't, but this is the exact format of the response you get from the microservice. Now that the app is integrated it needs to follow the same format, I'll create a task to clean this up in both repo's at a later time..

File: src/providers/journal/journal.spec.ts (hunk 3 above)
Comment (ammarhaiderbjss) [reply src/providers/journal/journal.spec.ts]: done

File: tslint.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "rules": {
-    "max-line-length": [true, { "limit": 100, "ignore-pattern": "^import |^export {(.*?)}" }],
+    "max-line-length": [true, { "limit": 200, "ignore-pattern": "^import |^export {(.*?)}" }],
```
Comment (ataylorBJSS) [inline tslint.json]: 200 on max-line-length!!! 💣 💥

File: src/pages/journal/journal.ts [hunk 5]
```diff
@@ -22,7 +22,7 @@ export class JournalPage {
   ) {}
 
   ionViewDidLoad() {
-    this.journalProvider.getData('email').subscribe(data => {
+    this.journalProvider.getData('test@test.com').subscribe(data => {
```
Comment (mathieuhendey) [inline src/pages/journal/journal.ts]: Need to be wary of using resolvable email addresses as dummies; use example.org email addresses.

AWS starts charging extra for bounced emails, I know that won't matter in this case but just something to keep in mind. Might be worth changing it here just for best practices

File: src/providers/journal/journal.spec.ts [hunk 6]
```diff
@@ -15,22 +19,22 @@ describe('PROVIDER: journal - ', () => {
...
   });
 
-  it(' should return a some journal slots', () => {
-    provider.getData('email').subscribe(slots => {
-      expect(slots.length).toBe(3);
+  it(' should return some journal slots', () => {
+    provider.getData('test@test.com').subscribe(slots => {
```
Comment (mathieuhendey) [inline src/providers/journal/journal.spec.ts]: Use \*@example.org email addresses in tests

**Pull request merged**